src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataRpcExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Text.Json;
+using Google.Protobuf.Collections;
+using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+
+namespace Microsoft.Azure.Functions.Worker.Grpc.FunctionMetadata
+{
+    internal static class FunctionMetadataRpcExtensions
+    {
+        internal static MapField<string, BindingInfo> GetBindingInfoList(this IFunctionMetadata funcMetadata)
+        {
+            if (funcMetadata is RpcFunctionMetadata rpcFuncMetadata)
+            {
+                return rpcFuncMetadata.Bindings;
+            }
+
+            MapField<string, BindingInfo> bindings = new MapField<string, BindingInfo>();
+            var rawBindings = funcMetadata.RawBindings;
+
+            foreach (var bindingJson in rawBindings)
+            {
+                var binding = JsonSerializer.Deserialize<JsonElement>(bindingJson);
+                BindingInfo bindingInfo = CreateBindingInfo(binding);
+                binding.TryGetProperty("name", out JsonElement jsonName);
+                bindings.Add(jsonName.ToString()!, bindingInfo);
+            }
+
+            return bindings;
+        }
+
+        private static BindingInfo CreateBindingInfo(JsonElement binding)
```

Comment:
nit - Wondering why the modifier here private and that of GetBindingInfoList is internal?

Reply:
This method was only used in this class (by `GetBindingInfoList`) but you've just made me realize I could cut down on some code elsewhere by making this internal and letting another class use it :)